fastavro/reader.py

```diff
@@ -375,10 +471,17 @@ def read_data(fo, writer_schema, reader_schema=None):
     """Read data from file object according to schema."""
 
     record_type = extract_record_type(writer_schema)
+    logical_type = extract_logical_type(writer_schema)
```

Comment:
In this case we have to pot two different classes of parses into one dispatcher. Keeping them separated makes things simpler.